lib/Terrain.cpp

```diff
@@ -50,204 +59,413 @@ Terrain::Manager::Manager()
 		JsonNode terrs(mod, ResourceID("config/terrains.json"));
 		for(auto & terr : terrs.Struct())
 		{
-			Terrain::Info info;
-			info.moveCost = terr.second["moveCost"].Integer();
+			auto * info = new TerrainType(terr.first); //set name
```

Comment:
What is specific reason to store raw pointers in objects? You still have memory leak BTW

Reply:
I can actually store them by value and handle by reference, tried and it works.